runtime-modules/content/src/nft/types.rs

```diff
@@ -169,160 +154,237 @@ impl<MemberId> Default for NftOwner<MemberId> {
+        self,
+        starting_block: BlockNumber,
+        last_bid_block: BlockNumber,
+    ) -> Self {
+        if last_bid_block.saturating_add(self.extension_period)
+            >= starting_block.saturating_add(self.auction_duration)
+        {
+            Self {
+                // bump auction duration when bid is made during extension period.
+                auction_duration: self.auction_duration.saturating_add(self.extension_period),
+                ..self
+            }
+        } else {
+            self
+        }
+    }
+}
+
+/// Open auction details
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub struct OpenAuctionDetails<BlockNumber: BaseArithmetic + Copy + Default> {
+    // bid lock duration
+    pub bid_lock_duration: BlockNumber,
+}
+
+/// Parameters used to issue a nft
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct NftIssuanceParametersRecord<MemberId, InitTransactionalStatus> {
+    /// Roayalty used for the author
+    pub royalty: Option<Royalty>,
+    /// Metadata
+    pub nft_metadata: NftMetadata,
+    /// member id Nft will be issued to
+    pub non_channel_owner: Option<MemberId>,
+    /// Initial transactional status for the nft
+    pub init_transactional_status: InitTransactionalStatus,
+}
+
+pub type NftIssuanceParameters<T> = NftIssuanceParametersRecord<
+    <T as common::MembershipTypes>::MemberId,
+    InitTransactionalStatus<T>,
+>;
+
+/// Initial Transactional status for the Nft: See InitialTransactionalStatusRecord above
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum InitTransactionalStatusRecord<
+    BlockNumber: BaseArithmetic + Copy + Default,
+    MemberId: Default + Copy + Ord,
+    Balance: Default + Clone + BaseArithmetic,
+    VideoId: Default + Copy,
+> {
+    Idle,
+    BuyNow(Balance),
+    InitiatedOfferToMember(MemberId, Option<Balance>),
+    Auction(AuctionParams<BlockNumber, Balance, MemberId, VideoId>),
+}
+
+impl<
+        BlockNumber: BaseArithmetic + Copy + Default,
+        MemberId: Default + Copy + Ord,
+        Balance: Default + Clone + BaseArithmetic,
+        VideoId: Default + Copy,
+    > Default for InitTransactionalStatusRecord<BlockNumber, MemberId, Balance, VideoId>
+{
+    fn default() -> Self {
+        Self::Idle
+    }
+}
+
+pub type InitTransactionalStatus<T> = InitTransactionalStatusRecord<
+    <T as frame_system::Trait>::BlockNumber,
+    <T as common::MembershipTypes>::MemberId,
+    CurrencyOf<T>,
+    <T as Trait>::VideoId,
+>;
+
+// Auction implementation
+/// Information on the auction being created.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct Bid<BlockNumber: BaseArithmetic + Copy + Default, Balance> {
```

Comment:
Should we actually split this into two types?

- `OpenAuctionBid`:
  - you do need a `auction_id` field, which connects the bid to a given auction, so that when owner is accepting bids, it has to be for the currently active auction.
- `EnglishAuctionBid`
  - you don't need the `made_at_block` I think.
  - you don't need to store it in the storage map `BidByAuctionByMember`, because you only need to store the last best bid inlined, like `last_bid` now, etc.
  
There is probably more here, but you get the general idea. Things are now working so differently across the two auction types that we presumably would raise the safety and clarity of our code a lot by embracing that.